orangecontrib/spectroscopy/widgets/owmultifile.py

```diff
@@ -41,6 +42,38 @@ def numpy_union_keep_order(A, B):
     return np.concatenate((A, to_add))
 
 
+def decimals_neeeded_for_unique_str(l):
+    min_diff = min(np.abs(np.diff(sorted(l))))
+    log_diff = math.log(min_diff, 10)
+    decimals = max(0, math.ceil(-log_diff))
```

Comment:
I hope this is fine. :) Could anyone find a counterexample?

Reply:
I woke up with an edge case where this would not work and have fixed the code accordingly.